packages/relay/src/services/ReportService.ts

```diff
@@ -280,6 +295,389 @@ export class ReportService {
+        reportId: string,
+        repType: ReputationType,
+        change: RepChangeType,
+        report: ReportHistory
+    ) {
+        res.status(200).json({
+            message: {
+                txHash,
+                reportId,
+                epoch: report.reportEpoch,
+                epochKey: report.respondentEpochKey,
+                type: repType,
+                score: change,
+            },
+        })
+    }
+
+    private handleError(
+        res: express.Response,
+        error: any,
+        direction: ReputationDirection
+    ) {
+        console.error(`Get ${direction} Reputation error:`, error)
+        if (error instanceof InternalError) {
+            res.status(error.httpStatusCode).json({
+                message: error.message,
+                error: error.message,
+            })
+        } else {
+            res.status(500).json({
+                message: `Get ${direction} Reputation error`,
+                error: error.message,
+                stack: error.stack,
+            })
+        }
+    }
+
+    private async updateReputationStatus(
+        db: DB,
+        type: ReputationDirection,
+        reportId: string,
+        nullifier?: string | undefined,
+        repUserType?: RepUserType
+    ) {
+        if (type === ReputationDirection.POSITIVE && nullifier) {
+            const report = await this.findReportWithNullifier(
+                db,
+                reportId,
+                nullifier,
+                ReportStatus.WAITING_FOR_TRANSACTION
+            )
+
+            if (report) {
+                const updatedAdjudicators = report.adjudicatorsNullifier.map(
+                    (adj) =>
+                        adj.nullifier === nullifier
+                            ? { ...adj, claimed: true }
+                            : adj
+                )
+
+                await db.update('ReportHistory', {
+                    where: { _id: report._id },
+                    update: {
+                        adjudicatorsNullifier: updatedAdjudicators,
+                    },
+                })
+            }
+        } else {
+            let updateField: string
+
+            if (type === ReputationDirection.POSITIVE) {
+                updateField = 'reportorClaimedRep'
+            } else {
+                if (repUserType === RepUserType.REPORTER) {
+                    updateField = 'reportorClaimedRep'
+                } else if (repUserType === RepUserType.POSTER) {
+                    updateField = 'respondentClaimedRep'
+                } else {
+                    throw new Error(
+                        'Invalid repUserType for negative reputation'
+                    )
+                }
+            }
+
+            const updateQuery = { [updateField]: true }
+            await db.update('ReportHistory', {
+                where: {
+                    reportId,
+                },
+                update: updateQuery,
+            })
+        }
+    }
```

Comment:
This implementation will do the wrong things below these situations:
- When `type` is `ReputationDirection.POSITIVE` but `repUserType` is `RepUserType.POSTER`, poster will claim postive reputation and prevent the reporter from claiming reputation.

Reply:
The status of the report has to be updated to completed if all users have claimed reputation.

Reply:
fixed